_posts/2024-06-26-performance-improvment-hybrid-query-215.md

```diff
@@ -0,0 +1,171 @@
+    - [In hybrid query optimize the way we iterate over results and collect scores of sub queries #745](https://github.com/opensearch-project/neural-search/issues/745)
+
+- **Reducing repetitive calculations:** To mitigate redundant internal computations, we have implemented value caching and reuse strategies, reducing the overall computational overhead within the system. By optimizing the handling of repetitive calculations and promoting value reuse, we have sped up the system by 5%. For more information, see the following GitHub issues:
+    - [Improve efficiency by caching and reusing internal calculations #756](https://github.com/opensearch-project/neural-search/issues/756)
+    - [Enhance performance through value caching and reuse strategies #764](https://github.com/opensearch-project/neural-search/issues/764)
+
+## Benchmark results
+
+Benchmark results show up to a 70% performance improvement for large datasets (over 10M) in hybrid queries with OpenSearch 2.15 compared to version 2.13. These benchmarks were conducted using a new OpenSearch Benchmark workload created specifically for evaluating semantic-search use cases. The following table summarizes benchmark results.
+
+<table>
+ <tr>
+  <th rowspan=2>Number of documents retrieved</th>
+  <th rowspan=2>Number of hybrid sub-queries</th>
+  <th colspan=3>OpenSearch version 2.13</th>
+  <th colspan=3>OpenSearch version 2.15</th>
+  <th>Performance improvement</th>
+ </tr>
+  <tr>
+  <th>p50, ms</th>
+  <th>p90, ms</th>
+  <th>p99, ms</th>
+  <th>p50, ms</th>
+  <th>p90, ms</th>
+  <th>p99, ms</th>
+  <th>%</th>
+ </tr>
+ <tr>
+  <td class="left"><b>1.6K</b></td>
+  <td>1</td>
+  <td>75</td>
+  <td>77</td>
+  <td>78</td>
+  <td>75</td>
+  <td>76</td>
+  <td>76</td>
+  <td class="light-green-clr bold">1</td>
+ </tr>
+ <tr>
+  <td class="left"><b>1.6M</b></td>
+  <td>1</td>
+  <td>224</td>
+  <td>240</td>
+  <td>245</td>
+  <td>109</td>
+  <td>114</td>
+  <td>119</td>
+  <td class="light-green-clr bold">52</td>
+ </tr>
+ <tr>
+  <td class="left"><b>10M</b></td>
+  <td>1</td>
+  <td>729</td>
+  <td>841</td>
+  <td>868</td>
+  <td>237</td>
+  <td>257</td>
+  <td>264</td>
+  <td class="light-green-clr bold">70</td>
+ </tr>
+  <tr>
+  <td class="left"><b>15M</b></td>
+  <td>3</td>
+  <td>1224</td>
+  <td>1300</td>
+  <td>1367</td>
+  <td>294</td>
+  <td>330</td>
+  <td>343</td>
+  <td class="light-green-clr bold">75</td>
+ </tr>
+ <tr>
+  <td colspan=8 class="center bold">% average boost in 2.15 vs 2.13</td>
+  <td class="light-green-clr bold">49</td>
+ </tr>
+</table>
+
+## Planned improvements
+
+We continue to analyze OpenSearch performance and identify opportunities for further enhancements. Future improvements may include:
```

Comment:
This reads as though a word should follow "performance". "performance KPIs"? "performance metrics"?

Reply:
I'll put "performance metrics"